extensions/amp-analytics/0.1/visibility-impl.js

```diff
@@ -155,27 +154,31 @@ export function isVisibilitySpecValid(config) {
  * @param {!String} selectionMethod The method to use to find the element..
  * @return {?Element} Element corresponding to the selector if found.
  */
-export function getElement(selector, el, selectionMethod) {
+export function getElement(ampdoc, selector, el, selectionMethod) {
   if (!el) {
     return null;
   }
 
   // Special case for root selector.
   if (selector == ':host' || selector == ':root') {
-    const elWin = el.ownerDocument.defaultView;
+    const elWin = ampdoc.win;
     const parentEl = elWin.frameElement && elWin.frameElement.parentElement;
     if (parentEl) {
       return closestBySelector(parentEl, '.-amp-element');
     }
   }
 
   if (selectionMethod == 'closest') {
+    // Restrict result to be contained by ampdoc.
+    if (!ampdoc.contains(el)) {
```

Comment:
I don't think this is what you need to do, unless I'm missing something. The input `el` is definitely in the ampdoc. However, the result of `closestByTag` may not be. Consider this:
```
result
  div
    - ampdoc-boundary - 
          container
             el
```

You are looking at `el` and then querying `closestByTag(el)` which yields `result`. So what just happened is that you traversed the ampdoc boundary by reaching to `result`. Thus all we need to do is to check that `ampdoc.contains(result)`.